lib/testLauncher/composeConfig.js

```diff
@@ -109,6 +108,52 @@ function findExtendConfigs(defaultConfigObject, extendPath, filePath, foundPaths
 	return mainConfigFile;
 }
 
+/**
+ * @description Search for configuration files by given path and filename without extension. Can search recursevly.
+ * @param {String} pathToSearch path to directory to search
+ * @param {String} filename base filename without extension
+ * @param {Boolean} shouldSearchDeep boolean value which shows is function should go back to 1 folder up. If true - goes up to the root directory.
+ */
+function deepDirectorySearch(pathToSearch, filename, shouldSearchDeep) {
+	const CONFIG_FORMATS = [
+		'.js',
+		'.json',
+		'.yaml',
+		'.yml',
+		'.json5',
+		'.ini',
+	];
+
+	let foundConfigFile = '';
+
+	if (fs.existsSync(pathToSearch)) {
+		const files = fs.readdirSync(pathToSearch);
+
+		files.forEach(file => {
```

Comment:
With `forEach` iteration will be happens for all entris even if value already found.
For avoid unnessesery work, please replace it by `find` function:
`foundConfigFile = files.find(file => {`